### PR TITLE
Fixing the location attribute added to mapInfoOp

### PR DESCRIPTION
**Pull request by anchuraj**
Named location attribute added to `tgt_offload_entry` shall be used by runtime calls like `ompx_dump_mapping_tables` to print the information of variables that are mapped to the device. `ompx_dump_mapping_tables` was printing the wrong location information and this change fixes it.

A sample execution of example before the change:
```
omptarget device 0 info: OpenMP Host-Device pointer mappings after block at libomptarget:0:0:

omptarget device 0 info: Host Ptr           Target Ptr         Size (B) DynRefCount HoldRefCount Declaration

omptarget device 0 info: 0x0000000000206df0 0x00007f02cdc00000 20000000 1           0            <program-file-loc> at unknown:18:35
```

The change replaces unknown to the mapped symbol and location to the declaration location.

### CODE REVIEW

File: flang/lib/Lower/OpenMP/OpenMP.cpp [hunk 1]
```diff
@@ -1644,11 +1644,14 @@ genTargetOp(Fortran::lower::AbstractConverter &converter,
         } else if (!fir::isa_builtin_cptr_type(eleType)) {
           mapFlag |= llvm::omp::OpenMPOffloadMappingFlags::OMP_MAP_TO;
           mapFlag |= llvm::omp::OpenMPOffloadMappingFlags::OMP_MAP_FROM;
-        }
-
+        };
+        auto context = converter.getFirOpBuilder().getContext();
```
Comment (agozillon) [inline flang/lib/Lower/OpenMP/OpenMP.cpp:1648]: Very minor nit (that I myself accidentally do all the time): should perhaps change the auto to the context's type (an MLIRContext I think) and perhaps use the firOpBuilder variable that exists already if it's possible as opposed to getting it through the converter! Or alternatively you could change the argument "context" given to NameLoc::get to "firOpBuilder.getContext()". The same applies to the above in the other function motionClauses that one day soon will be aligned as it's for the most part identical!

**Review by agozillon — APPROVED** [review]
Other than the small nit, this PR LGTM! Thank you for the patch :-)

File: flang/lib/Lower/OpenMP/OpenMP.cpp (hunk 1 above)
Comment (anchuraj) [reply flang/lib/Lower/OpenMP/OpenMP.cpp:1648]: Thank you for reviewing my PR. I missed the exiting variable. Implemented the suggestion

File: flang/lib/Lower/OpenMP/ClauseProcessor.cpp [hunk 2]
```diff
@@ -903,9 +903,12 @@ bool ClauseProcessor::processMap(
...
           // optimisation passes may alter this to ByCopy or other capture
           // types to optimise
+          auto location = mlir::NameLoc::get(
+              mlir::StringAttr::get(firOpBuilder.getContext(), asFortran.str()),
+              symAddr.getLoc());
           mlir::Value mapOp = createMapInfoOp(
-              firOpBuilder, clauseLocation, symAddr, mlir::Value{},
-              asFortran.str(), bounds, {},
+              firOpBuilder, location, symAddr, mlir::Value{}, asFortran.str(),
+              bounds, {},
```
Comment (skatrak) [inline flang/lib/Lower/OpenMP/ClauseProcessor.cpp:911]: Nit: Add comments to document hardcoded arguments, same comment for the other call to `createMapInfoOp` in OpenMP.cpp.
```suggestion
              firOpBuilder, location, symAddr, /*varPtrPtr=*/mlir::Value{}, asFortran.str(),
              bounds, /*members=*/{},
```

**Review by skatrak — APPROVED** [review]
Thank you Anchu, this LGTM as well!

File: flang/lib/Lower/OpenMP/ClauseProcessor.cpp (hunk 2 above)
Comment (anchuraj) [reply flang/lib/Lower/OpenMP/ClauseProcessor.cpp]: Hi @skatrak , Thank you for reviewing the change. This was introduced in a later commit. My merge from main , merged those changes

**Pull request merged**